### PR TITLE
Feature/dispatch error status

### PR DESCRIPTION
**Pull request by SilentGert**
I added the `status` code from the last PR now also to the dispatched `*_FAILURE` request actions so third-party applications can use it as well.
Furthermore, it now stored in the state as well.

In order to test it, I added [fetch-mock](http://www.wheresrhys.co.uk/fetch-mock/) which only mocks the `fetch` API as the name says and allows us to do proper integration tests. While writing the tests I removed some `.only()` which was limiting the test execution and fixed another integration test by using fetch-mock there as well.

### CODE REVIEW

File: test/specs/integration/pathFilter.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/**
+ * Returns a function which checks if a given URL starts
+ * with the previously provided path parameter.
+ */
+export default function pathFilter(path: string) {
+  return (matchingUrl: string) => matchingUrl.startsWith(path)
```
[the remark this answers is not in the document]
Comment (SilentGert) [reply test/specs/integration/pathFilter.js]: Uhh gotta check that out.

File: test/specs/integration/createDispatchActions.spec.js [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+      description: 'All the infos here',
+    })
+
+    setTimeout(() => {
+      expect(actions).to.have.length(2)
+
+      const elementId = actions[0].payload.elementId
+
+      expect(actions[0]).to.deep.equal({
+        type: 'KRAKEN_CREATE_DISPATCH',
```
[the remark this answers is not in the document]
Comment (SilentGert) [reply test/specs/integration/createDispatchActions.spec.js]: I see your point. What I like about the current solution is that it is explicit. So you cannot hide a bug / breaking change in the action creator without the test failing (dispatch a broken action and compare it to the same broken version here). However, it is more maintenance and this part could also easily be covered by unit tests for the action creators (if that is not already the case).

Main question back at you, what we want to achieve with the integration tests here?

[the remark this answers is not in the document]
Comment (frontendphil) [reply test/specs/integration/createDispatchActions.spec.js]: My focus would be that the payload of the action is handled correctly. For that, I don't need to know how actions are structured and other data they carry. That's why I like the solution where you use the action creator. This puts more focus on the payload (maybe that's just me reading the test). 

Also, as you pointed out it reduces the maintenance overhead and whether the action creator works correctly should be covered in another unit test.

[the remark this answers is not in the document]
Comment (SilentGert) [reply test/specs/integration/createDispatchActions.spec.js]: Alright, then I'll adjust the tests :)

**Comment by SilentGert** [comment]
Mr. fancy product manager aka @frontendphil feel free to review again.

**Pull request merged**